examples/account-federation-step-function/steps/add-to-close-io/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM node:16.15-stretch-slim as root
+# Install openssl in slim, allowing modules which require this to work.
+# Without this, Prisma etc. will not be able to connect to databases.
+RUN apt-get update && apt-get install -y openssl && rm -rf /var/lib/apt/lists/*
+
+FROM root AS builder
+
+# Create a build stage, which compiles TS to regular ol' JS.
+WORKDIR /opt/
+COPY package*.json *.lock tsconfig.json .
+RUN npm install
+COPY . .
+RUN npm run build
+
+# This is the output image, which contains only the prebuilt JS.
+FROM root
+WORKDIR /opt/
+COPY package*.json /opt/
+RUN npm install --production
```

Comment:
All of these examples are using Yarn (`yarn.lock`), but then install deps using `npm`. I think this will result in us creating a new `package-lock.json` at build time which means our deps could shift.

We could either:

- Keep `yarn.lock` and add `yarn install --frozen-lockfile` here (we might also need to install Yarn)
- Replace `yarn.lock` with `package-lock.json` and add `npm ci` here

Personal preference would be to stick to npm as it ships with Node so is the assumed default, but I don't actually know npm vs. Yarn usage numbers.